src/client/init/root.tsx

```diff
@@ -19,34 +19,53 @@ function useHydrated() {
 
 export const defineClientConfig = (config: RdtClientConfig) => config
 
-export const withDevTools = (Component: any, config?: RemixDevToolsProps) => () => {
-	hydrationDetector()
-	const hydrated = useHydrated()
-	if (!hydrated) return <Component />
-
-	return (
-		<>
-			<Component />
-			{createPortal(<RemixDevTools {...config} />, document.body)}
-		</>
-	)
-}
-
-/**
- *
- * @description Injects the dev tools into the Vite App, ONLY meant to be used by the package plugin, do not use this yourself!
- */
-export const withViteDevTools = (Component: any, config?: RemixDevToolsProps) => () => {
-	hydrationDetector()
-	function AppWithDevTools(props: any) {
+export const withDevTools =
+	(Component: any, config?: RemixDevToolsProps & { panelMode?: "auto" | "embedded" }) => () => {
+		hydrationDetector()
 		const hydrated = useHydrated()
+		const isEmbedded = useIsEmbeddedRDT()
 		if (!hydrated) return <Component />
+
+		if ((config?.panelMode ?? "auto") === "embedded") {
+			return (
+				<RDTEmbeddedContextProvider>
+					<Component />
+				</RDTEmbeddedContextProvider>
+			)
+		}
+
 		return (
 			<>
-				<Component {...props} />
```

Comment:
not spreading the props over the component actually breaks RR7 support, I learned it the hard way yesterday with line 27 above breaking my rr7 test setup. So props need to be accepted and spread across Component in every return case

Reply:
Also, this withDevTools method is kind of low-key deprecated

Reply:
> deprecated

in favor of?

Reply:
WithViteDevTools, this one is undocumented relic used to manually set it up if needed in remix v1, I plan to remove it fully in next major